src/main/java/org/broadinstitute/hellbender/engine/VariantWalker.java

```diff
@@ -31,6 +33,28 @@ public abstract class VariantWalker extends VariantWalkerBase {
     private FeatureDataSource<VariantContext> drivingVariants;
     private FeatureInput<VariantContext> drivingVariantsFeatureInput;
 
+    @Argument(fullName = StandardArgumentDefinitions.VARIANT_OUTPUT_INTERVAL_FILTERING_MODE_LONG_NAME,
+            doc = "Restrict the output variants to ones that match the specified intervals according to the specified matching mode.",
+            optional = true)
+    @Advanced
+    public IntervalFilteringVcfWriter.Mode outputVariantIntervalFilteringMode  = getDefaultVariantOutputFilterMode();
+
+    /**
+     * @return Default interval filtering mode for variant output.  Subclasses may override this to set a different default.
+     */
+    public IntervalFilteringVcfWriter.Mode getDefaultVariantOutputFilterMode(){
+        return IntervalFilteringVcfWriter.Mode.ANYWHERE;
+    }
+
+    @Override
+    public IntervalFilteringVcfWriter.Mode getVariantFilteringOutputModeIfApplicable() {
+        if (outputVariantIntervalFilteringMode != null) {
+            return outputVariantIntervalFilteringMode;
+        } else {
+            return super.getVariantFilteringOutputModeIfApplicable();
```

Comment:
Why the if null then super here?  It seems unnecessary?

Reply:
oh whoops. missed a null, made it distinguish however...